### DISCUSSION TITLE
What is the best way to check if the size of a store is higher than a particular number?

### DISCUSSION THREAD

**Discussion by ate47** [eclipse-rdf4j/rdf4j#3734] [Q&A] [answered]
I have a native store and I want to check if my number of triples is higher than a particular number X, should I use size() >= X or a getStatements(null, null, null, false) and test to see if I have at least X statements in the returned iterator.

I've tried both with X = 4,

```java
        try (SailConnection connection = nativeStore.getConnection()) {
            int i;
            StopWatch timer = new StopWatch();
            try (CloseableIteration<? extends Statement, SailException> it = connection.getStatements(null, null, null, false)) {
                for (i = 0; i < 4 && it.hasNext(); i++) {
                    it.next();
                }
            }
            System.out.println((i == 4) + ", " + timer.stopAndShow());
            timer.reset();

            System.out.println((connection.size() >= 4) + ", " + timer.stopAndShow());

            System.out.println(connection.size());
        }
```

And the output result is: 

```
true, 584 us // with getStatements(...)
true, 27 ms 771 us // with size()
80000
```

With those results I want to use ``getStatements(...)``, but before I want to know if this is always the best scenario or if it exists a 3rd solution?

Thanks,

ATE

**Comment by hmottestad** [comment, marked as answer]
Size method typically uses getStatements, but needs to iterate over all the statements to count them. Since you are checking for an upper limit, then it will typically be faster to use getStatements the way you are doing because you can stop iterating once you hit your limit. 

The size method could potentially have a faster implementation on some stores or under certain circumstances, but I don't that we are currently doing anything special here.

**Reply by ate47** [reply]
Thanks! I forgot to answer to you yesterday.
